core/src/main/java/io/grpc/internal/PickFirstLeafLoadBalancer.java

```diff
@@ -181,21 +178,30 @@ public Status acceptResolvedAddresses(ResolvedAddresses resolvedAddresses) {
     return Status.OK;
   }
 
-  private static EquivalentAddressGroup removeDuplicateAddresses(EquivalentAddressGroup eag) {
-    Set<SocketAddress> addressSet = new HashSet<>();
-    ArrayList<SocketAddress> addrs = new ArrayList<>(); // maintains order
-
-    for (SocketAddress address : eag.getAddresses()) {
-      if (addressSet.add(address)) {
-        addrs.add(address);
-      }
+  private static List<EquivalentAddressGroup> deDupAddresses(List<EquivalentAddressGroup> groups) {
+    if (groups == null) {
```

Comment:
There's no way groups can be null here. `addresses` is required in `ResolvedAddresses` and we've already dereferenced it multiple times.

Reply:
removed